crypto/decryption.go

```diff
@@ -170,3 +171,46 @@ func decryptReaderTo(readerRaw io.Reader, writer io.Writer, decrypter cipher.Blo
 
 	return nil
 }
+
+func ConvertToSpki(pemB64PublicKey string) (string, error) {
+	// Decode from base64
+	pemPublicKey, err := base64.StdEncoding.DecodeString(pemB64PublicKey)
+
+	if err != nil {
+		glog.Errorf("Error decoding base64 encoded key: %v", err)
+		return "", err
+	}
+
+	block, _ := pem.Decode([]byte(pemPublicKey))
+	if block == nil {
+		err := fmt.Errorf("failed to parse PEM block containing the public key")
+		return "", err
+	}
+
+	pub, err := x509.ParsePKCS1PublicKey(block.Bytes)
+	if err != nil {
+		glog.Errorf("failed to parse RSA public key: %v", err)
+		return "", err
+	}
+
+	pubBytes, err := x509.MarshalPKIXPublicKey(pub)
+	if err != nil {
+		glog.Errorf("failed to marshal RSA public key: %v", err)
+		return "", err
+	}
+
+	pubPEM := pem.EncodeToMemory(&pem.Block{
+		Type:  "PUBLIC KEY",
+		Bytes: pubBytes,
+	})
+
+	pubPEMStr := string(pubPEM)
+
+	// Remove header and footer
+	spkiPubKeyBase64 := strings.Join(strings.Split(pubPEMStr, "\n")[1:len(strings.Split(pubPEMStr, "\n"))-2], "")
```

Comment:
A couple things:
 - You are splitting by `\n` and then joining with an empty string. Is this correct? If so, can you update the comment to explain that as well?

Cosmetic:
 - I was confused by this variable name given what the comment says, maybe just update `pubPEMStr` here?
 - Could you break up this transformation in a couple of lines? It's a bit hard to understand what's happening here and there are duplicate function calls. Sth like a helper `lines := strings.Join(strings.Split(pubPEMStr, "\n")` would already help a lot!

Reply:
I split it into two lines and made an explainatory comment! Let me know if it's better!